homeassistant/components/sense/__init__.py

```diff
@@ -39,34 +44,92 @@
 )
 
 
-async def async_setup(hass, config):
-    """Set up the Sense sensor."""
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Sense component."""
+    hass.data.setdefault(DOMAIN, {})
+    conf = config.get(DOMAIN)
+    if not conf:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_EMAIL: conf[CONF_EMAIL],
+                CONF_PASSWORD: conf[CONF_PASSWORD],
+                CONF_TIMEOUT: conf.get(CONF_TIMEOUT),
+            },
+        )
+    )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Sense from a config entry."""
 
-    username = config[DOMAIN][CONF_EMAIL]
-    password = config[DOMAIN][CONF_PASSWORD]
+    entry_data = entry.data
+    email = entry_data[CONF_EMAIL]
+    password = entry_data[CONF_PASSWORD]
+    timeout = entry_data[CONF_TIMEOUT]
 
-    timeout = config[DOMAIN][CONF_TIMEOUT]
     try:
-        hass.data[SENSE_DATA] = ASyncSenseable(api_timeout=timeout, wss_timeout=timeout)
-        hass.data[SENSE_DATA].rate_limit = ACTIVE_UPDATE_RATE
-        await hass.data[SENSE_DATA].authenticate(username, password)
+        gateway = ASyncSenseable(api_timeout=timeout, wss_timeout=timeout)
```

Comment:
If this cannot raise, let's move it outside of the `try`.

Reply:
It can only raise if you provide a username and password.  will move it out